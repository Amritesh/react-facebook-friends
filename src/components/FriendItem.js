import React from 'react';
import Label from './Label';
import Radium from 'radium';
import { POST as pType, LIKE as lType, COMMENT as cType } from '../utils/constants';
import { facebookBlue } from '../utils/constants';

const colors = {
  0: '#DB2828',
  1: '#F2711C',
  2: '#FBBD08',
  3: '#B5CC18',
  4: '#21BA45',
  5: '#00B5AD',
  6: '#2185D0',
  7: '#6435C9',
  8: '#A333C8',
  9: '#E03997',
  10: '#A5673F',
  11: '#767676',
  12: '#1B1C1D'
};

const style = {
  item: {
    margin: '5px auto',
    maxWidth: '750px',
    backgroundColor: 'white',
    borderBottom: '1px solid #eee',
    '@media (max-width: 48em)': {
      position: 'relative'
    }
  },
  anchor: {
    padding: '1em 1em',
    display: 'block',
    color: 'black',
    ':hover': {
      backgroundColor: '#f7f7f7'
    }
  },
  rank: {
    position: 'relative',
    top: '-30px',
    marginRight: '10px',
    textAlign: 'center',
    fontWeight: '400',
    fontSize: '4em',
    width: '80px',
    display: 'inline-block',
    color: 'white',
    borderRadius: '50%'
  },
  imgWrapper: {
    padding: '10px',
    display: 'inline-block'
  },
  img: {
    border: '1px solid rgb(59, 89, 152)',
    borderRadius: '3px'
  },
  name: {
    textAlign: 'center',
    color: facebookBlue,
    fontWeight: 'bold',
    fontSize: '0.8em'
  },
  clear: {
    clear: 'both'
  },
  label: {
    textAlign: 'center'
  },
  value: {
    fontSize: '6em'
  }
};

const labelMap = {
  c: {
    text: 'Comments',
    icon: 'fa fa-comments',
    top: '63px'
  },
  l: {
    text: 'Likes',
    icon: 'fa fa-thumbs-o-up',
    top: '26px'
  },
  p: {
    text: 'Posts',
    icon: 'fa fa-pencil',
    top: '100px'
  }
};

const FriendItem = ({ name, rank, link, url, LIKE, COMMENT, POST }) => (
  <div style={style.item}>
    <a target="_blank" href={link} style={style.anchor}>
      <div style={{
        ...style.rank,
        backgroundColor: colors[(rank - 1) % 10]
      }}>{rank}</div>
      <div style={style.imgWrapper}>
        <img src={url} alt={name} style={style.img} />
        <div style={style.name}>{name}</div>
      </div>
      <Label {...labelMap.l} value={LIKE} />
      <Label {...labelMap.c} value={COMMENT} />
      <Label {...labelMap.p} value={POST} />
      <div style={style.clear}></div>
    </a>
  </div>
);

export default Radium(FriendItem);
