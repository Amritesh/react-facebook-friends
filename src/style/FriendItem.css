.anchor {
  min-width: 300px;
  border-radius: 3px;
  margin-top: 5px;
  margin-bottom: 5px;
  background: white;
  max-width: 680px;
  padding: 1px 15px;
  text-decoration: none;
  display: flex;
  align-items: center;
  min-height: 150px;
  justify-content: space-around;
  color: black;
}

.anchor:hover {
  background: #f7f7f7;
}

.rank {
  text-align: center;
  font-weight: 400;
  font-size: 4em;
  width: 80px;
  display: inline-block;
  color: white;
  border-radius: 50%;
}

.imgWrapper {
  padding: 10px;
}

.img {
  border: 1px solid rgb(59, 89, 152);
  border-radius: 3px;
}

.name {
  text-align: center;
  color: rgb(58, 87, 149);
  font-weight: bold;
  font-size: .8em;
}

.label {
  text-align: center;
}

.value {
  font-size: 6em;
}

.labelWrapper {
  display: flex;
}

@media (max-width: 590px) {
  .labelWrapper {
    flex-wrap: wrap;
    min-height: inherit;
    align-items: center;
  }
}