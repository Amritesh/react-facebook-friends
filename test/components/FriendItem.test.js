import React from 'react';
import renderer from 'react-test-renderer';
import FriendItem from '../../src/components/FriendItem';

test('FriendItem component should renders correctly', () => {
  const props = {
    name: 'name',
    rank: 1,
    link: 'link',
    url: 'url',
    LIKE: 1,
    COMMENT: 2,
    POST: 3
  };
  const component = renderer.create(<FriendItem {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
