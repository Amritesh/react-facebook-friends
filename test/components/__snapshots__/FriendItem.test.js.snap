// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendItem component should renders correctly 1`] = `
<a
  className="anchor"
  href="link"
  rel="noopener noreferrer"
  target="_blank"
>
  <div
    className="rank"
    style={
      Object {
        "backgroundColor": "#DB2828",
      }
    }
  >
    1
  </div>
  <div
    className="imgWrapper"
  >
    <img
      alt="name"
      className="img"
      src="url"
    />
    <div
      className="name"
    >
      name
    </div>
  </div>
  <div
    className="labelWrapper"
  >
    <div
      className="label-wrapper"
    >
      <div
        className="label-valueGroup"
      >
        <i
          className="fa fa-thumbs-o-up"
        />
        <span>
          1
        </span>
      </div>
      <div
        className="label-text"
      >
        Likes
      </div>
    </div>
    <div
      className="label-wrapper"
    >
      <div
        className="label-valueGroup"
      >
        <i
          className="fa fa-comments"
        />
        <span>
          2
        </span>
      </div>
      <div
        className="label-text"
      >
        Comments
      </div>
    </div>
    <div
      className="label-wrapper"
    >
      <div
        className="label-valueGroup"
      >
        <i
          className="fa fa-pencil"
        />
        <span>
          3
        </span>
      </div>
      <div
        className="label-text"
      >
        Posts
      </div>
    </div>
  </div>
</a>
`;
